### DISCUSSION TITLE
No backwards compatibility for string concatinations

### DISCUSSION THREAD

**Discussion by alfie-nsugh** [modular/modular#2189] [Q&A] [answered]
I was wondering if it was intended that this would give an error:
  1> x = "h" 
  2. x =x+  "ello" 
  3. print(x) 
[User] Failed to produce standalone archive: failed to lower module to LLVM IR for archive compilation
[User] Expression [1] wrapper:17:9: note: see current operation: %60 = "pop.string.concat"(%59, %23) : (!kgen.string, !kgen.string) -> !kgen.string

[User] error: Expression [1]:2:5: failed to legalize operation 'pop.string.concat' that was explicitly marked illegal
x =x+  "ello"

When doing 
```
var z: String = "H"
z += "ello"
```

it does work. However, in python we don't have to declare something to be mutable to be able to change it. 
Is this departure from python intended or are you planning to change this in the future?

**Comment by rarebreed** [comment]
It shouldn't be an issue with mutability.  Firstly, mojo got rid of `let`, so essentially everything on the stack is mutable (sadly) and there are only immutable references.  Also, not declaring with `var` implicitly makes it a var, and inside a `def` function, gives it python function level scoping.  But I think this is an issue with the REPL, not with mutability.

The code you have works fine in a script.

```mojo
def main():
    z = "h"
    z = z + "ello"
    print(z)
```

**Comment by hogepodge** [comment, marked as answer]
Hi @alfie-nsugh this is a known issue with string literals in the REPL. We have an issue, and are working on fixing it.
